xarray/core/dataarray.py

```diff
@@ -3505,6 +3508,202 @@ def pad(
         )
         return self._from_temp_dataset(ds)
 
+    def idxmin(
+        self,
+        dim: Hashable = None,
+        skipna: bool = None,
+        fill_value: Any = dtypes.NA,
+        keep_attrs: bool = None,
+        **kwargs: Any,
```

Comment:
What other arguments would be valid to be passed as `kwargs`?

I would be inclined to list these specifically rather than handling anything, which could unintentionally expose details of xarray's internal implementation as a public API.

Reply:
This is here because `argmin` and `argmax` have it and I wanted to keep as close to them as possible.  These arguments are passed directly to `argmin` and `argmax`.  As long as our `argmin` and `argmax` support it I would be inclined to keep it here.

Reply:
argmin/argmax have this because they use a generic template for reduction functions, but if we wrote more specific wrappers for them I would also drop `**kwargs`.

Reply:
I can remove it if you really want.

Reply:
Yes, I'd prefer to keep it out for now. It's always easier to expand API later rather shrink them :)

Reply:
Okay